src/types/misc/CommunityModel.jl

```diff
@@ -0,0 +1,136 @@
+"""
+$(TYPEDSIGNATURES)
+
+Shallow copy of a [`CommunityModel`](@ref)
+"""
+Base.copy(m::CommunityModel) = CommunityModel(;
+    members = m.members,
+    abundances = m.abundances,
+    environmental_links = m.environmental_links,
+)
+
+"""
+$(TYPEDSIGNATURES)
+
+Shallow copy of a [`CommunityModel`](@ref)
+"""
+Base.copy(m::CommunityMember) = CommunityMember(;
+    id = m.id,
+    model = m.id,
+    exchange_reaction_ids = m.exchange_reaction_ids,
+    biomass_reaction_id = m.biomass_reaction_id,
+)
+
+"""
+$(TYPEDSIGNATURES)
+
+Shallow copy of a [`EnvironmentalLink`](@ref)
+"""
+Base.copy(m::EnvironmentalLink) = EnvironmentalLink(;
+    reaction_id = m.reaction_id,
+    metabolite_id = m.metabolite_id,
+    lower_bound = m.lower_bound,
+    upper_bound = m.upper_bound,
+)
+
+"""
+$(TYPEDSIGNATURES)
+
+A helper function that creates an exchange/environmental variable linking matrix
+for community member `m`.
+"""
+function env_ex_member_matrix(
+    m::CommunityMember,
+    env_mets::Vector{String},
+    env_rxns::Vector{String},
+)
+    idxs = [
+        (i, j) for
+        (i, j) in enumerate(indexin(env_rxns, variables(m.model))) if !isnothing(j)
+    ]
+    sparse(
+        first.(idxs),
+        last.(idxs),
+        ones(length(idxs)),
+        length(env_mets),
+        n_variables(m.model),
+    )
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+A helper function that creates the entire exchange/environmental variable
+linking matrix for a community model.
+"""
+function env_ex_matrix(cm)
```

Comment:
type (also, can we have a sensible name like environment_exchange_stoichiometry or so)

Reply:
Fixed